### PR TITLE
SOLR-15824 Improved Query Screen raw query parameters section

### PR DESCRIPTION
**Pull request by betulince**
https://issues.apache.org/jira/browse/SOLR-15824

# Description

While sending a query in solr, it is very convenient that the q field can be resizable(because it's in textarea tag), while the fq and raw query parameters fields are not resizable, which can cause difficulties in long query parameters. To solve this made improvement on query.html and query.css.

# Solution

I put fq and raw query parameters inside textarea tag and with css I made it vertical resizable because of the line between query and result. Because if it was resizable both vertical and horizontal or only horizontal, it would cause the appearance of the panel to deteriorate.

# Checklist

Please review the following and check all that apply:

- [X] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [X] I have created a Jira issue and added the issue ID to my pull request title.
- [X] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [X] I have developed this patch against the `main` branch.
- [X] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

----------------------------------------------------------------------------------------------------------------------------------------------------------------------
# EDIT

I changed only raw query parameters section because of the comments on PR. Added +/- buttons for raw query parameters.

### CODE REVIEW

**Comment by epugh** [comment]
Just gave it a try and it looks great!   A bit of extra space makes a nice bit of difference.

image

Also, I didn't know that if I put `fq=manu_id_s:corsair`, in the raw params, they get yanked up to the fq section~  cool.

**Comment by epugh** [comment]
Does having each fq make sense to be a text area?  Do they actually get really long when they get split up like this:

image

**Comment by epugh** [comment]
Maybe we have a text area that starts out as two rows intesead of three?

**Comment by epugh** [comment]
This is a nice improvement, how would you like to be credited in CHANGES.txt?

**Comment by thelabdude** [comment]
I think it would be nice to have a + / - list for raw params like we have for `fq` ... perhaps with a toggle between a list of text boxes or the textarea added to this PR? The textarea is an improvement for sure, but with a long list, it might still be better to separate each parameter out in its own line

**Comment by epugh** [comment]
That makes sense to me @thelabdude...   what do you think @betulince ???

**Comment by betulince** [comment]
Hello,

It's a really good idea to have +/- for raw parameters like in fq, I will
work on it!!

And I would like to be on CHANGES.txt, it would make me more motivated to
work on UI/UX for admin page or query screen!

And about having fq textarea, I will talk about it with my company because
we are using Solr everyday in our company. I will see if it's worth to do
it or not.

**Comment by betulince** [comment]
What do you mean ? It's already 2 rows

Eric Pugh ***@***.***>, 2 Ara 2021 Per, 18:40 tarihinde şunu
yazdı:

**Comment by epugh** [comment]
> What do you mean ? It's already 2 rows Eric Pugh ***@***.***>, 2 Ara 2021 Per, 18:40 tarihinde şunu yazdı:
For some reason it looked like three in my browser!!!   

However, I think that @thelabdude suggestion of instead using the +/- is a better route than the text area!

**Comment by thelabdude** [comment]
My suggestion was to provide the option of + / - or the resizable text area vs. one or the other ... at some point even the resizable text area will be hard to see all the details if there are enough parameters. Be cool if I could paste in a long query string of params, have the JS parse them when I switch over to the +/- ... I know this form pretty well from adapting it for the query tester in Schema Designer. I can take a stab at an implementation there to show the concept.

**Comment by betulince** [comment]
> Just gave it a try and it looks great! A bit of extra space makes a nice bit of difference.
> 
> image
> 
> Also, I didn't know that if I put `fq=manu_id_s:corsair`, in the raw params, they get yanked up to the fq section~ cool.

I took back the changes on fq section based on your comments. There is no need to add resizable area when we already have +/- buttons for parameters.

**Comment by betulince** [comment]
> > What do you mean ? It's already 2 rows Eric Pugh _**@**_.***>, 2 Ara 2021 Per, 18:40 tarihinde şunu yazdı:
> > For some reason it looked like three in my browser!!!
> 
> However, I think that @thelabdude suggestion of instead using the +/- is a better route than the text area!

Added +/- buttons for raw query parameters. Tests are done. There is no problem about sending and executing queries and taking results as you can see from screenshot:


Screenshot from 2021-12-06 21-59-51

**Comment by betulince** [comment]
> This is a nice improvement, how would you like to be credited in CHANGES.txt?

Of course!!

**Comment by betulince** [comment]
Hello,

I sent another commit which I changed the improvement based on feedbacks, I
am waiting for you to check so I can move forward with another improvements.

Timothy Potter ***@***.***>, 4 Ara 2021 Cmt, 01:17 tarihinde
şunu yazdı:

**Comment by thelabdude** [comment]
I'll kick the tires on this now.

**Comment by thelabdude** [comment]
I suspect this is probably an existing bug (not introduced by the changes in this PR) (see screenshot), but if you supply the same parameter more than once, such as `stats.field` (to get stats for multiple fields), then the values get concatenated into a single comma-delimited list sent to the server like `"stats.field":"pages_i,price"`. 
Screen Shot 2021-12-10 at 2 08 47 PM

I'd like to see this bug fixed as part of this PR if you're up to it @betulince ? If not, we can tackle under another change / JIRA. I suspect is b/c the form is getting submitted 2x everytime I click the `Execute Query` button?!? The first time the query succeeds but then it re-submits with `"stats.field":"pages_i,price"` which fails.
Screen Shot 2021-12-10 at 2 47 52 PM

Also, line 51 is calling `addRawParams(urlParams[p]);` which doesn't exist?

**Comment by thelabdude** [comment]
Confirmed ... the query form getting submitted 2x each time you push the `Execute Query` button exists on main! So that's bad and looks to be the cause of why having `stats.field=f1&stats.field=f2` fails (b/c of how the params are processed for the 2nd query)

**Comment by betulince** [comment]
I'll fix what you mentioned above with this PR.

**Comment by betulince** [comment]
> I suspect this is probably an existing bug (not introduced by the changes in this PR) (see screenshot), but if you supply the same parameter more than once, such as `stats.field` (to get stats for multiple fields), then the values get concatenated into a single comma-delimited list sent to the server like `"stats.field":"pages_i,price"`. Screen Shot 2021-12-10 at 2 08 47 PM
> 
> I'd like to see this bug fixed as part of this PR if you're up to it @betulince ? If not, we can tackle under another change / JIRA. I suspect is b/c the form is getting submitted 2x everytime I click the `Execute Query` button?!? The first time the query succeeds but then it re-submits with `"stats.field":"pages_i,price"` which fails. Screen Shot 2021-12-10 at 2 47 52 PM
> 
> Also, line 51 is calling `addRawParams(urlParams[p]);` which doesn't exist?

**Comment by betulince** [comment]
I closed PR by accident, fyi.

**Comment by betulince** [comment]
> Confirmed ... the query form getting submitted 2x each time you push the `Execute Query` button exists on main! So that's bad and looks to be the cause of why having `stats.field=f1&stats.field=f2` fails (b/c of how the params are processed for the 2nd query) ... I think you need to have rawParams for this 2x behavior to happen.
Screenshot from 2021-12-12 13-29-39

The query form isn't getting submitted 2x everytime I click the Execute Query button anymore.

**Comment by betulince** [comment]
> I suspect this is probably an existing bug (not introduced by the changes in this PR) (see screenshot), but if you supply the same parameter more than once, such as `stats.field` (to get stats for multiple fields), then the values get concatenated into a single comma-delimited list sent to the server like `"stats.field":"pages_i,price"`. Screen Shot 2021-12-10 at 2 08 47 PM
> 
> I'd like to see this bug fixed as part of this PR if you're up to it @betulince ? If not, we can tackle under another change / JIRA. I suspect is b/c the form is getting submitted 2x everytime I click the `Execute Query` button?!? The first time the query succeeds but then it re-submits with `"stats.field":"pages_i,price"` which fails. Screen Shot 2021-12-10 at 2 47 52 PM
> 
> Also, line 51 is calling `addRawParams(urlParams[p]);` which doesn't exist?

Screenshot from 2021-12-12 13-47-38

Fixed.

**Comment by betulince** [comment]
bug solved, fyi.

can you check if it's giving another errors ?

Timothy Potter ***@***.***>, 11 Ara 2021 Cmt, 00:55 tarihinde
şunu yazdı:

**Comment by epugh** [comment]
@betulince I put two small comments in, and this code looks great!  Learned some stuff about Angular ;-)

**Comment by betulince** [comment]
'Betul Ince' is good.

Thank you!!

**Comment by epugh** [comment]
@betulince I went ahead and made the changes that I suggested, and want to push to your branch, can you give me permissions?  I think it's a  checkbox to "allow committers to push to branch" or something like that!   I'll get the changelog updated and this merged!!!

**Comment by betulince** [comment]
> @betulince I went ahead and made the changes that I suggested, and want to push to your branch, can you give me permissions? I think it's a checkbox to "allow committers to push to branch" or something like that! I'll get the changelog updated and this merged!!!

You already committed I guess!

File: solr/webapp/web/js/angular/controllers/query.js [hunk 1]
```diff
@@ -176,18 +183,21 @@ solrAdminApp.controller('QueryController',
...
-      Query.query(params, function(data) {
-        $scope.lang = $scope.val['wt'];
-        if ($scope.lang == undefined || $scope.lang == '') {
-          $scope.lang = "json";
-        }
-        $scope.response = data;
-        // Use relative URL to make it also work through proxies that may have a different host/port/context
-        $scope.url = url;
-        $scope.hostPortContext = $location.absUrl().substr(0,$location.absUrl().indexOf("#")); // For display only
+      if (isPageReload) {
```
Comment (thelabdude) [inline solr/webapp/web/js/angular/controllers/query.js:186]: awesome! good fix here

**Review by thelabdude — APPROVED** [review]
looks good to me

**Pull request merged**